feat: supporting Consistency Group drilldowns in snapmirror dashboard

Hardikl opened the pull request.

Hardikl commented:
As clusters are in bigtop, creating the system there and update for review.

cgrinds commented on cmd/collectors/rest/plugins/snapmirror/snapmirror.go:
```diff
@@ -182,9 +185,14 @@ func (my *SnapMirror) updateSMLabels(data *matrix.Matrix) {
+		// find cgName from the destination_location
+		matches := cgNameRe.FindStringSubmatch(cgInstance.GetLabel("destination_location"))
+		if len(matches) == 2 {
+			cgInstance.SetLabel("cg_name", matches[1])
```
I wonder if this would work as well and be clearer?
cgInstance.SetLabel("cg_name", filepath.Base(cgInstance.GetLabel("destination_location")))

Hardikl replied:
yes, much cleaner.

Hardikl commented:
PR Changes includes these:
- Added variable for consistency group 
- Added panel of `Consistency Group` in Highlights which shows count of cg and can be filtered by CG var
- Added `Consistency Group Data Protection Analysis per Cluster` panels which can be filtered by CG var.

Hardikl commented on grafana/dashboards/cmode/snapmirror.json:
```diff
@@ -345,7 +345,7 @@
       "cacheTimeout": null,
       "datasource": "${DS_PROMETHEUS}",
-      "description": "Number of insync snapmirrors.",
+      "description": "Number of consistency group protected snapmirrors.",
```
Sure, We can discuss this in meeting.

Hardikl commented on grafana/dashboards/cmode/snapmirror.json:
```diff
@@ -379,6 +379,78 @@
         "x": 12,
         "y": 3
       },
+      "id": 23,
```
Yes, handled.

Hardikl commented:
Merged 2 tables for snapmirror
<img width="1863" alt="image" src="https://github.com/NetApp/harvest/assets/83282894/12fca9b6-cb69-4576-ac19-2a9c9ad46032">


Merged 2 tables for CG
<img width="1866" alt="image" src="https://github.com/NetApp/harvest/assets/83282894/398900b5-7963-423f-94b3-aaadcf934230">


pending item: 
- filtering in highlights table based on CG var

Hardikl commented:
Supporting `Top $TopResources Last Transfers by Transfer Data Size` table filter with CG var,

<img width="1728" alt="image" src="https://github.com/NetApp/harvest/assets/83282894/7c1b36cb-5bac-411e-9f0b-c04d95a0788e">

Just to note, 
There are 3 queries in table and below graphs, `snapmirror_last_transfer_duration`, `snapmirror_last_transfer_size` and `snapmirror_lag_time`. If any point in time the metrics won't be available for that relationship(there may be many reason as no lag or no successful transfer or unhealthy relationhsip, etc..), then it won't be visible in the table/graph.

Hardikl commented:
After hide the node vars, validated that all the panels were populated as expected after hiding them.

<img width="1735" alt="image" src="https://github.com/NetApp/harvest/assets/83282894/10c7c2ce-2187-43b4-ae6b-8167b6052853">

Hardikl commented:
Added `sourceConsistencyGroup` and `DestinationConsistencyGroup` variables.

<img width="1727" alt="image" src="https://github.com/NetApp/harvest/assets/83282894/2187f8b3-fb46-48ac-8898-5a08a98704e8">

Hardikl commented:
Updated the UI panels to accommodate the flexgroup panel count.

<img width="1719" alt="image" src="https://github.com/NetApp/harvest/assets/83282894/b69d7c9d-ca9a-4431-a921-e86b46b2ce85">

Added decent documentation of all kind of relationships.

<img width="492" alt="image" src="https://github.com/NetApp/harvest/assets/83282894/27510e12-5457-45e8-8dc9-a5ebfebe7682">

cgrinds commented:
Looks good @Hardikl. @Nikhita-13 has also checked and verified these changes.
